dist-config/src/main/java/org/keycloak/config/Option.java

```diff
@@ -0,0 +1,64 @@
+package org.keycloak.config;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class Option<T> {
+
+    public enum Runtime {
+        QUARKUS,
+        DOCS,
+        OPERATOR
+    }
+
+    private final Class<T> type;
+    private final String key;
+    private final OptionCategory category;
+    private final Set<Runtime> hidden;
+    private final boolean build;
+    private final String description;
+    private final T defaultValue;
```

Comment:
Should we make this an optional? Not every configuration has a default Value atm

Reply:
No strong opinions, I do believe that we are mostly using `null`s in this codebase though.

Reply:
done, at the end